### PR TITLE
Hide malformed parameters from error page

### LINKED ISSUE #29947: Unable to parse `filter_parameters` in development environment leads to unclear error message
When malformed query parameters are passed, a controller fails to parse the query parameters (which is expected). In production, this is ok because the `PublicExceptions` middleware catches it and shows the corresponding error page. In development on the other hand, the `DebugExceptions` middleware attempts to render the [request and response template](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/middleware/templates/rescues/_request_and_response.html.erb#L9) but is unable to parse `filtered_parameters` and doesn't display the correct error message. 

Here is the test that replicates the bug:

```diff
diff --git a/railties/test/application/middleware/exceptions_test.rb b/railties/test/application/middleware/exceptions_test.rb
index fe07ad3cbe..8ad218beb8 100644
--- a/railties/test/application/middleware/exceptions_test.rb
+++ b/railties/test/application/middleware/exceptions_test.rb
@@ -134,5 +134,21 @@ def index
       assert_match(/boooom/, last_response.body)
       assert_match(/測試テスト시험/, last_response.body)
     end
+
+    test "displays diagnostics message when malformed query parameters are provided" do
+      controller :foo, <<-RUBY
+        class FooController < ActionController::Base
+          def index
+          end
+        end
+      RUBY
+
+      app.config.action_dispatch.show_exceptions = true
+      app.config.consider_all_requests_local = true
+
+      get "/foo?x[y]=1&x[y][][w]=2"
+      assert_equal 400, last_response.status
+      assert_match "Rack::QueryParser::ParameterTypeError", last_response.body
+    end
   end
 end
```

If you run the test above, it'll fail with the message below:

```
$ rails -v
Rails 5.2.0.alpha
$ ruby -v
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
$ ruby -I"lib:test" test/application/middleware/exceptions_test.rb
Run options: --seed 20823

# Running:

......F

Failure:
ApplicationTests::MiddlewareExceptionsTest#test_displays_diagnostics_message_when_malformed_query_parameters_are_provided [test/application/middleware/exceptions_test.rb:151]:
Expected /Rack::QueryParser::ParameterTypeError/ to match "".
```

### PR DESCRIPTION
**Pull request by st0012**
TLDR; If we don't hide malformed parameters, Rails will get an error when trying to render debugging page and the page won't be displayed properly.

This is a continuation of #32209 with some improvements and fixes #29947

### CODE REVIEW

**Comment by st0012** [comment]
ping @sikachu

File: actionpack/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ## Rails 6.0.0.rc1 (April 24, 2019) ##
 
+*   Fix a bug where DebugExceptions errors out when malformed query parameters are provided
+
+    *Yuki Nishijima*, *Stan Lo*
+
```
Comment (eileencodes) [inline actionpack/CHANGELOG.md]: This should be above the title header since this change will be for 6.0.0.rc2 not 6.0.0.rc1

File: actionpack/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 ## Rails 6.0.0.rc1 (April 24, 2019) ##
 
+*   Fix a bug where DebugExceptions errors out when malformed query parameters are provided
```
Comment (eileencodes) [inline actionpack/CHANGELOG.md]:
```suggestion
*   Fix a bug where DebugExceptions throws an error when malformed query parameters are provided.
```

File: actionpack/lib/action_dispatch/middleware/debug_view.rb [hunk 3]
```diff
@@ -56,5 +56,13 @@ def render(*)
     def protect_against_forgery?
       false
     end
+
+    def params_readable?
```
Comment (eileencodes) [inline actionpack/lib/action_dispatch/middleware/debug_view.rb]: `params_readable?` to me reads like "can be read" rather than "is not malformed". I'm struggling to come up with a better name though.

**Review by eileencodes — CHANGES_REQUESTED** [review]
Requesting changes for the changelog.

For the method name, I'm not sure I like `params_readable?` but not sure what a better name would be.

File: actionpack/test/dispatch/debug_exceptions_test.rb [hunk 4]
```diff
@@ -620,4 +620,23 @@ def call(env)
...
+  test "debug exceptions app shows diagnostics when malformed query parameters are provided" do
+    @app = DevelopmentApp
+
+    get "/bad_request?x[y]=1&x[y][][w]=2"
+
+    assert_response 400
+    assert_match "ActionController::BadRequest", body
+  end
+
+  test "debug exceptions app shows diagnostics when malformed query parameters are provided by xhr" do
```
Comment (abhaynikam) [inline actionpack/test/dispatch/debug_exceptions_test.rb]: As per https://guides.rubyonrails.org/api_documentation_guidelines.html#wording . I think it should be written as `xhr` -> `XHR`.
 
```suggestion
  test "debug exceptions app shows diagnostics when malformed query parameters are provided by XHR" do
```

**Review by eileencodes — APPROVED** [review]
Looks good, can you squash your commits into 1?

**Comment by st0012** [comment]
@eileencodes done!

**Comment by st0012** [comment]
@eileencodes this change is based on `6-0-stable`, should I cherry pick the commit to master? Or you'll do it?

**Comment by eileencodes** [comment]
Oh I missed the base branch was incorrect. Next time this should go straight to master and be cherry-picked into 6-0-stable instead.

**Comment by st0012** [comment]
@eileencodes I see, thanks!
